typer/core.py

```diff
@@ -656,6 +657,7 @@ def main(
         complete_var: Optional[str] = None,
         standalone_mode: bool = True,
         windows_expand_args: bool = True,
+        rich_markup_mode: MarkupMode = None,
```

Comment:
I don't think this argument should be added to the `main` function - instead we should use `self.rich_markup_mode`. I'll experiment with this and push to your branch directly.

Reply:
Fixing this also resolved some of the coverage issues.